lib/utils/compose.js

```diff
@@ -227,10 +227,7 @@ function originalTarDirectory(dir, param) {
 				fs.stat(file),
 				readFile(file),
 				(filename, stats, data) =>
-					pack.entry(
-						{ name: toPosixPath(filename), size: stats.size, mode: stats.mode },
-						data,
-					),
+					pack.entry({ ...stats, name: toPosixPath(filename) }, data),
```

Comment:
OK. I worry a bit that `...stats` might includes properties that would inadvertently invalidate the cache (the opposite of what we are trying to achieve) if developers push to the same balena app from different machines (whether the same developer of different developers), because it includes lots of properties:
```
Stats {
  dev: 2114,
  ino: 48064969,
  mode: 33188,
  nlink: 1,
  uid: 85,
  gid: 100,
  rdev: 0,
  size: 527,
  blksize: 4096,
  blocks: 8,
  atimeMs: 1318289051000.1,
  mtimeMs: 1318289051000.1,
  ctimeMs: 1318289051000.1,
  birthtimeMs: 1318289051000.1,
  atime: Mon, 10 Oct 2011 23:24:11 GMT,
  mtime: Mon, 10 Oct 2011 23:24:11 GMT,
  ctime: Mon, 10 Oct 2011 23:24:11 GMT,
  birthtime: Mon, 10 Oct 2011 23:24:11 GMT }
```
But if you have tested it, I am OK with it. :-)